### PR TITLE
In Html::renderSelectOptions() pass 3 argument not by reference  + Adopt psalm.

### PR DESCRIPTION
**Pull request by vjik**
| Q             | A
| ------------- | ---
| Is bugfix?    |  ❌
| New feature?  | ❌
| Breaks BC?    | ✔️
| Fixed issues  | -

- In Html::renderSelectOptions() pass 3 argument not by reference
- Fix: incorrect work with iterable $selection in methods checkboxList, radioList and renderSelectOptions
- Update dev dependencies
- Adopt Psalm (error level 1)
- Clean up

### CODE REVIEW

File: src/Html.php [hunk 1]
```diff
@@ -1384,21 +1527,45 @@ public static function ol($items, array $options = []): string
...
      *
+     * @psalm-param iterable<array-key, string>|string|null $selection
+     * @psalm-param array<array-key, array|string> $items
+     *
      * @throws JsonException
      *
      * @return string the generated list options
      */
-    public static function renderSelectOptions($selection, array $items, array &$tagOptions = []): string
+    public static function renderSelectOptions($selection, array $items, array $tagOptions = []): string
```
Comment (samdark) [inline src/Html.php:1537]: What's the reson for this change?

Comment (vjik) [reply src/Html.php:1537]: Usability. Without this change 3th argument should be variable, can't use method like this: 
`Html::renderSelectOptions($selection, $items, ['encode' => false])`

File: src/Html.php [hunk 2]
```diff
@@ -1103,17 +1211,28 @@ public static function checkboxList(string $name, $selection = null, array $item
         $name = self::getArrayableName($name);
 
         if (is_iterable($selection)) {
-            $selection = array_map('strval', (array)$selection);
+            $selection = array_map('strval', is_array($selection) ? $selection : iterator_to_array($selection));
```
Comment (samdark) [inline src/Html.php:1214]: Originally it meant that if a single value is selected then you can specify it without wrapping it into array explicitly. Now it changed its meaning.

File: src/Html.php [hunk 3]
```diff
@@ -1181,24 +1301,46 @@ public static function checkboxList(string $name, $selection = null, array $item
...
+     *
      * @throws JsonException
      *
      * @return string the generated radio button list
      */
     public static function radioList(string $name, $selection = null, array $items = [], array $options = []): string
     {
         if (is_iterable($selection)) {
-            $selection = array_map('strval', (array)$selection);
+            $selection = array_map('strval', is_array($selection) ? $selection : iterator_to_array($selection));
```
Comment (samdark) [inline src/Html.php:1322]: Originally it meant that if a single value is selected then you can specify it without wrapping it into array explicitly. Now it changed its meaning.

File: src/Html.php [hunk 4]
```diff
@@ -1384,21 +1549,49 @@ public static function ol($items, array $options = []): string
...
+     *
+     * @throws JsonException
+     *
+     * @return string
+     */
+    private static function renderSelectOptionTags($selection, array $items, array &$tagOptions): string
     {
         if (is_iterable($selection)) {
-            $selection = array_map('strval', (array)$selection);
+            $selection = array_map('strval', is_array($selection) ? $selection : iterator_to_array($selection));
```
Comment (samdark) [inline src/Html.php:1581]: Originally it meant that if a single value is selected then you can specify it without wrapping it into array explicitly. Now it changed its meaning.

**Review by samdark — CHANGES_REQUESTED** [review]

File: src/Html.php (hunk 2 above)
Comment (vjik) [reply src/Html.php:1214]: No... this is only for iterable values and originally this correctly work with `ArrayObject` and not correctly with custom iterable object. For example, this test fail originally:

```php
Html::checkboxList('test', new IterableObject(['value2']), $this->getDataItems(), [
	'item' => static function ($index, $label, $name, $checked, $value) {
		return $index . Html::label($label . ' ' . Html::checkbox($name, $checked, ['value' => $value]));
	},
	'tag' => false,
]);
```

Now  single value as selected also work good.

**Review by samdark — APPROVED** [review]

**Comment by samdark** [comment]
👍

**Pull request merged**